pkg/generator/generator.go

```diff
@@ -42,40 +65,39 @@ func Generate(typename string, keyType string, slice bool, wd string) error {
 	return nil
 }
 
-func getData(typeName string, keyType string, slice bool, wd string) (templateData, error) {
+func getData(valueType *ValueType, keyType string, wd string) (templateData, error) {
 	var data templateData
-	parts := strings.Split(typeName, ".")
-	if len(parts) < 2 {
-		return templateData{}, fmt.Errorf("type must be in the form package.Name")
-	}
-
-	name := parts[len(parts)-1]
-	importPath := strings.Join(parts[:len(parts)-1], ".")
 
 	genPkg := getPackage(wd)
 	if genPkg == nil {
 		return templateData{}, fmt.Errorf("unable to find package info for " + wd)
 	}
 
+	name := valueType.Name
+
 	data.Package = genPkg.Name
 	data.LoaderName = name + "Loader"
 	data.BatchName = lcFirst(name) + "Batch"
 	data.Name = lcFirst(name)
 	data.KeyType = keyType
-	data.Slice = slice
+	data.Slice = valueType.IsSlice
 
-	prefix := "*"
-	if slice {
+	prefix := ""
+	if valueType.IsSlice {
 		prefix = "[]"
 		data.LoaderName = name + "SliceLoader"
```

Comment:
Name generation probably also needs to be updated to handle these new cases, or let the user specify a name in case there are collisions.